src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -373,33 +402,29 @@ private static bool TryParseExactD(ReadOnlySpan<char> guidString, ref GuidResult
                 return false;
             }
 
-            ref Guid g = ref result._parsedGuid;
-
-            if (TryParseHex(guidString.Slice(0, 8), out Unsafe.As<int, uint>(ref g._a)) && // _a
+            if (TryParseHex(guidString.Slice(0, 8), out result._a) && // _a
                 TryParseHex(guidString.Slice(9, 4), out uint uintTmp)) // _b
             {
-                g._b = (short)uintTmp;
+                result._b = (ushort)uintTmp;
 
                 if (TryParseHex(guidString.Slice(14, 4), out uintTmp)) // _c
                 {
-                    g._c = (short)uintTmp;
+                    result._c = (ushort)uintTmp;
 
                     if (TryParseHex(guidString.Slice(19, 4), out uintTmp)) // _d, _e
                     {
-                        g._d = (byte)(uintTmp >> 8);
-                        g._e = (byte)uintTmp;
+                        // _d, _e must be stored as a big-endian ushort
+                        result._de = (BitConverter.IsLittleEndian) ? BinaryPrimitives.ReverseEndianness((ushort)uintTmp) : (ushort)uintTmp;
```

Comment:
```suggestion
                        result._de = BitConverter.IsLittleEndian ? BinaryPrimitives.ReverseEndianness((ushort)uintTmp) : (ushort)uintTmp;
```

Nit - multiple places. I believe the usual style for src/libraries is without the extra `( )` in this context.

`(BitConverter.IsLittleEndian) ?` - 27 occurences under src/libraris
`BitConverter.IsLittleEndian ?` - 3 occurrences under src/libraries

Reply:
Maybe this is an argument to add `BinaryPrimitives.htonl`, which would encapsulate the check and make the call sites cleaner? Right now I think this API sits on `IPAddress`, which is the wrong layering.